accounts/abi/bind/backends/simulated.go

```diff
@@ -36,12 +40,15 @@ var chainConfig = &core.ChainConfig{HomesteadBlock: big.NewInt(0)}
 // This nil assignment ensures compile time that SimulatedBackend implements bind.ContractBackend.
 var _ bind.ContractBackend = (*SimulatedBackend)(nil)
 
+var errBlockNumberUnsupported = errors.New("SimulatedBackend cannot access blocks other than the latest block")
```

Comment:
Why do you think it defeats it's purpose?

Defining an error and *not* export it is a totally sane decision. In most cases we define them such that we may reuse the same error message in several places without having to create a `errors.New` every time we want to return an error.

In addition to that we do usually not rely on error types directly and we don't treat them as exceptions either.